tests/connectionpool_test.go

```diff
@@ -137,3 +143,193 @@ func TestConnectionPool(t *testing.T) {
 		return err
 	}))
 }
+
+// TestConnectionPoolWithReuseInboundUser verifies that a ConnectionPool can be created without
+// specifying a username, which enables the "Reuse Inbound User" feature. This allows the
+// connection pool to use the credentials of whoever is connecting to the pool rather than
+// a fixed service user.
+func TestConnectionPoolWithReuseInboundUser(t *testing.T) {
```

Comment:
Suggestion: WDYT about testing that the actual "Reuse Inbound User" feature too? 

[NEX-1317-test-suggestion.patch](https://github.com/user-attachments/files/19385495/NEX-1317-test-suggestion.patch)